lib/react_client/react_interop.dart

```diff
@@ -34,8 +36,83 @@ abstract class React {
 
   external static bool isValidElement(dynamic object);
   external static ReactClass get Fragment;
+
+  external static JsRef createRef();
+}
+
+/// Creates a [Ref] object that can be attached to an element via the ref attribute.
+///
+/// __Example__:
+///
+///     class FooComponent extends react.Component2 {
+///       final Ref<BarComponent> barRef = createRef();
+///       final Ref<InputElement> inputRef = createRef();
+///
+///       render() => react.div({}, [
+///         Bar({'ref': barRef}),
+///         react.input({'ref': inputRef}),
+///       ]);
+///     }
+///
+/// Learn more: <https://reactjs.org/docs/refs-and-the-dom.html#creating-refs>.
+Ref<CurrentType> createRef<CurrentType>() {
+  return new Ref<CurrentType>();
+}
+
+/// The return type of [createRef], the constructor wraps [React.createRef] for use in
+/// Dart.
+///
+/// A [Ref] can be attached to an element via the ref attribute. When the element renders,
+/// a reference to that node becomes accessible at the [current] attribute of the [Ref].
+///
+/// See: <https://reactjs.org/docs/refs-and-the-dom.html#creating-refs>.
+class Ref<CurrentType> {
+  JsRef jsRef;
+
+  Ref() {
+    jsRef = React.createRef();
+  }
+
+  /// When the element renders, a reference to that node becomes accessible at the
+  /// [current] attribute of the [Ref].
```

Comment:
Related to above style guide tips, as well as "PREFER starting library or type comments with noun phrases." https://dart.dev/guides/language/effective-dart/documentation#prefer-starting-variable-getter-or-setter-comments-with-noun-phrases
 this comment could be something more like:
```dart
/// A reference to the latest instance of the rendered component.
///
/// See: <https://reactjs.org/docs/refs-and-the-dom.html#creating-refs>.
```